__tests__/integration/hathorwallet_others.test.js

```diff
@@ -0,0 +1,1235 @@
+        index: expect.any(Number),
+        address: expect.any(String),
+        authorities: TOKEN_MINT_MASK,
+      }]);
+    expect(hWallet.selectAuthorityUtxo(tokenHash, wallet.isMeltOutput.bind(wallet), { many: true }))
+      .toStrictEqual([{
+        tx_id: tokenHash,
+        index: expect.any(Number),
+        address: expect.any(String),
+        authorities: 2,
+      }]);
+
+    // Validating the wrapper method
+    expect(hWallet.getAuthorityUtxos(tokenHash, 'mint'))
+      .toStrictEqual([{
+        tx_id: tokenHash,
+        index: expect.any(Number),
+        address: expect.any(String),
+        authorities: TOKEN_MINT_MASK,
+      }]);
+    expect(hWallet.getAuthorityUtxos(tokenHash, 'melt'))
+      .toStrictEqual([{
+        tx_id: tokenHash,
+        index: expect.any(Number),
+        address: expect.any(String),
+        authorities: 2,
+      }]);
+  });
+
+  it('should find many "mint" authority utxos', async () => {
+    // Delegating the mint to another address on the same wallet
+    const mintDelegationTx = await hWallet.delegateAuthority(
+      tokenHash,
+      'mint',
+      hWallet.getAddressAtIndex(1),
+      { createAnother: false }
+    );
+    await waitForTxReceived(hWallet, mintDelegationTx.hash);
+
+    // Should not find the spent utxo
+    expect(hWallet.selectAuthorityUtxo(tokenHash, wallet.isMintOutput.bind(wallet)))
+      .toStrictEqual([{
+        tx_id: mintDelegationTx.hash,
+        index: expect.any(Number),
+        address: expect.any(String),
+        authorities: TOKEN_MINT_MASK,
+      }]);
+    expect(hWallet.selectAuthorityUtxo(tokenHash, wallet.isMintOutput.bind(wallet), { many: true }))
+      .toStrictEqual([{
+        tx_id: mintDelegationTx.hash,
+        index: expect.any(Number),
+        address: expect.any(String),
+        authorities: TOKEN_MINT_MASK,
+      }]);
+    expect(hWallet.getAuthorityUtxos(tokenHash, 'mint'))
+      .toStrictEqual([{
+        tx_id: mintDelegationTx.hash,
+        index: expect.any(Number),
+        address: expect.any(String),
+        authorities: TOKEN_MINT_MASK,
+      }]);
+
+    // Should return multiple utxos
+    expect(hWallet.selectAuthorityUtxo(
+      tokenHash,
+      wallet.isMintOutput.bind(wallet),
+      { many: true, skipSpent: false }
+    ))
+      .toStrictEqual([
+        {
+          tx_id: tokenHash,
+          index: expect.any(Number),
+          address: expect.any(String),
+          authorities: TOKEN_MINT_MASK,
+        },
+        {
+          tx_id: mintDelegationTx.hash,
+          index: expect.any(Number),
+          address: expect.any(String),
+          authorities: TOKEN_MINT_MASK,
+        },
+      ]);
+  });
+
+  it('should find many "melt" authority utxos', async () => {
+    // Delegating the mint to another address on the same wallet
+    const meltDelegationTx = await hWallet.delegateAuthority(
+      tokenHash,
+      'melt',
+      hWallet.getAddressAtIndex(1),
+      { createAnother: false }
```

Comment:
I would try to do this differently with melt. Instead of finding one spent and one unspent, why don't you create two melt authority utxos and find both in the `hWallet.selectAuthorityUtxo(tokenHash, wallet.isMeltOutput.bind(wallet), { many: true })`? Just to test something new.

Reply:
Good idea, implemented this on [f97a9954](https://github.com/HathorNetwork/hathor-wallet-lib/pull/406/commits/f97a9954c06758a57da6f1f9bb24a43392690914).